presto-kudu/src/main/java/io/prestosql/plugin/kudu/KuduClientSession.java

```diff
@@ -543,7 +543,11 @@ private KuduPredicate createComparisonPredicate(ColumnSchema columnSchema, KuduP
         io.prestosql.spi.type.Type type = TypeHelper.fromKuduColumn(columnSchema);
         Object javaValue = TypeHelper.getJavaValue(type, value);
         if (javaValue instanceof Long) {
-            return KuduPredicate.newComparisonPredicate(columnSchema, op, (Long) javaValue);
+            if (type.getClass().getSuperclass().equals(DecimalType.class)) {
```

Comment:
What if we could handle in `TypeHelper` instead of here, In the line 158 https://github.com/prestosql/presto/blob/a6e7ab26195732cb3821b32443bf96a780d163a2/presto-kudu/src/main/java/io/prestosql/plugin/kudu/TypeHelper.java#L158

Reply:
That make sense, I can return a BigDecimal(nativeValue) in TypeHelper#getJavaValue when type instanceof DecimalType is trye, and at KuduClientSession#createComparisonPredicate I can handle this value directly.
You are right!